### ISSUE TITLE
How generatePng method of class AspectModelDiagramGenerator in esmf-aspect-model-document-generators module, stores JF<<random_number>>.tmp file in tmp 

### ISSUE THREAD

**Issue by richaashara** [eclipse-esmf/esmf-sdk#370]
In module **esmf-aspect-model-document-generators**,  there is a class named AspectModelDiagramGenerator has a method called **generatePng**, which creates a JF<<random_number>>.tmp file (example: +~JF10887631845317785145.tmp) in the tmp folder. But, I am not seeing any underneath implementation for storing generated content. Could someone help me with how it does that?

**Comment by atextor** [comment]
Hi @richaashara ,
the file generated in the temp folder are technical artifacts that are caused by how the generation is done: Diagrams are created using [graphviz](https://graphviz.org/) and this process may or may not (depending on your operating system) create a temporary file.

Usage of the Java SDK API is documented in [Java Tooling for Working with Aspect Models](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/java-aspect-tooling.html), I suggest in particular the sections on [Loading and Saving Aspect Models](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/java-aspect-tooling.html#loading-and-saving) followed by [Generating SVG or PNG Diagrams](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/java-aspect-tooling.html#generating-diagrams).

In short, you'd do something like this:
```java
VersionedModel model = AspectModelResolver.loadAndResolveModel( modelFile )
  .getOrElseThrow(() -> new RuntimeException("put some better error handling here"));
AspectModelDiagramGenerator generator = new AspectModelDiagramGenerator( model );
generator.generateDiagram( AspectModelDiagramGenerator.Format.SVG, Locale.ENGLISH,
  new FileOutputStream( "/some/place/diagram.svg" ) );
```

**Comment by richaashara** [comment]
I am using this SDK inside [semantic-hub](https://github.com/eclipse-tractusx/sldt-semantic-hub). I have replaced the existing SDK with this one. It generates these tmp files, the name starting with JF.  But those files are not getting deleted and  I am running the app as a docker container so files are getting stored in container memory. After every request container memory consumption increases.  
I have debugged the request. it seems the below method of Font class(package: package java.awt), is not deleting this tmp file as the flag copiedFontData will be always true in case no exception is raised. I am using java 17

`    private static Font[] createFont0(int fontFormat, InputStream fontStream, boolean allFonts, CreatedFontTracker tracker) throws FontFormatException, IOException {
        if (fontFormat != 0 && fontFormat != 1) {
            throw new IllegalArgumentException("font format not recognized");
        } else {
            boolean copiedFontData = false;

            try {
                final File tFile = (File)AccessController.doPrivileged(new PrivilegedExceptionAction<File>() {
                    public File run() throws IOException {
                        return Files.createTempFile("+~JF", ".tmp").toFile();
                    }
                });
                if (tracker != null) {
                    tracker.add(tFile);
                }

                int totalSize = 0;

                try {
                    OutputStream outStream = (OutputStream)AccessController.doPrivileged(new PrivilegedExceptionAction<OutputStream>() {
                        public OutputStream run() throws IOException {
                            return new FileOutputStream(tFile);
                        }
                    });
                    if (tracker != null) {
                        tracker.set(tFile, outStream);
                    }

                    try {
                        byte[] buf = new byte[8192];

                        while(true) {
                            int bytesRead = fontStream.read(buf);
                            if (bytesRead < 0) {
                                break;
                            }

                            if (tracker != null) {
                                if (totalSize + bytesRead > 33554432) {
                                    throw new IOException("File too big.");
                                }

                                if (totalSize + tracker.getNumBytes() > 335544320) {
                                    throw new IOException("Total files too big.");
                                }

                                totalSize += bytesRead;
                                tracker.addBytes(bytesRead);
                            }

                            outStream.write(buf, 0, bytesRead);
                        }
                    } finally {
                        outStream.close();
                    }

                    copiedFontData = true; // expllictely set true
                    FontManager var24 = FontManagerFactory.getInstance();
                    Font2D[] font2DArr = var24.createFont2D(tFile, fontFormat, allFonts, true, tracker);
                    int num = font2DArr.length;
                    Font[] fonts = new Font[num];

                    for(int i = 0; i < num; ++i) {
                        fonts[i] = new Font(font2DArr[i]);
                    }

                    Font[] var26 = fonts;
                    return var26;
                } finally {
                    if (tracker != null) {
                        tracker.remove(tFile);
                    }

                    if (!copiedFontData) { // condtion will not satisfy in case of success
                        if (tracker != null) {
                            tracker.subBytes(totalSize);
                        }

                        class NamelessClass_1 implements PrivilegedExceptionAction<Void> {
                            NamelessClass_1() {
                            }

                            public Void run() {
                                tFile.delete();
                                return null;
                            }
                        }

                        AccessController.doPrivileged(new NamelessClass_1());
                    }

                }
            } catch (Throwable var22) {
                if (var22 instanceof FontFormatException) {
                    throw (FontFormatException)var22;
                } else if (var22 instanceof IOException) {
                    throw (IOException)var22;
                } else {
                    Throwable cause = var22.getCause();
                    if (cause instanceof FontFormatException) {
                        throw (FontFormatException)cause;
                    } else {
                        throw new IOException("Problem reading font data.");
                    }
                }
            }
        }
    }`
